pybossa/api/task_run.py

```diff
@@ -70,14 +72,23 @@ def _preprocess_post_data(self, data):
         project_id = data['project_id']
         self.check_can_post(project_id, task_id)
         preprocess_task_run(project_id, task_id, data)
-        info = data['info']
-        path = "{0}/{1}/{2}".format(project_id, task_id, current_user.id)
         if with_encryption:
-            data['info'] = {
-                'pyb_answer_url': upload_json_data(json_data=info, upload_path=path,
-                    file_name='pyb_answer.json', encryption=with_encryption,
-                    conn_name='S3_TASKRUN', upload_root_dir=upload_root_dir)
-            }
+            info = data['info']
+            path = "{0}/{1}/{2}".format(project_id, task_id, current_user.id)
+
+            # for tasks with private_json_encrypted_payload, generate
+            # encrypted response payload under private_json_encrypted_response
+            encrypted_response = encrypt_task_response_data(task_id, project_id, info)
+            data['info'] = {}
+            pyb_answer_url = upload_json_data(json_data=info, upload_path=path,
+                file_name='pyb_answer.json', encryption=with_encryption,
+                conn_name='S3_TASKRUN', upload_root_dir=upload_root_dir
+            )
+            if pyb_answer_url:
+                data['info']['pyb_answer_url'] = pyb_answer_url
+            if encrypted_response:
+                data['info']['private_json_encrypted_response'] = encrypted_response
```

Comment:
Looks good, just confirming the naming of the field should use single underscores or a double.

`private_json_encrypted_response` versus `private_json__encrypted_response`.

Reply:
This is going to be reserved keyword that would be documented. Hence, either single/double underscores can be used.